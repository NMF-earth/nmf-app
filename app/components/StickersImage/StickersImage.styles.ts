import { StyleSheet } from "react-native";

import { screen } from "../../constants/Layout";

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  image: {
    marginVertical: 20,
    alignSelf: "center",
    maxWidth: 320,
    width: "100%",
    height: screen.width - 100,
  },
});

export default styles;
