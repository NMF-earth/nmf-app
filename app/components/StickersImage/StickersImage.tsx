import React from "react";
import { View, Image } from "react-native";

import styles from "./StickersImage.styles";
import ImagesAssets from "../../constants/ImagesAssets";

interface Prop {
  sticker: "restaurant" | "bike" | "earth";
}

const StickersImage = ({ sticker }: Prop) => {
  let source = ImagesAssets.stickers.earth;

  if (sticker === "restaurant") {
    source = ImagesAssets.stickers.restaurant;
  } else if (sticker === "bike") {
    source = ImagesAssets.stickers.bike;
  }

  return (
    <View style={styles.container}>
      <Image style={styles.image} resizeMode="contain" source={source} />
    </View>
  );
};

export default StickersImage;
