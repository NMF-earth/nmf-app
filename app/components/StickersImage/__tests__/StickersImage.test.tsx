import React from "react";
import { create } from "react-test-renderer";

import StickersImage from "../";

jest.unmock("../");

it("renders correctly StickersImage with bike image", () => {
  const tree = create(<StickersImage sticker="bike" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly StickersImage with restaurant image", () => {
  const tree = create(<StickersImage sticker="restaurant" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly StickersImage with earth image", () => {
  const tree = create(<StickersImage sticker="earth" />).toJSON();
  expect(tree).toMatchSnapshot();
});
