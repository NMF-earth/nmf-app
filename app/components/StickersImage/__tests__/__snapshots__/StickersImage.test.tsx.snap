// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly StickersImage with bike image 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <Image
    resizeMode="contain"
    source={1}
    style={
      {
        "alignSelf": "center",
        "height": 650,
        "marginVertical": 20,
        "maxWidth": 320,
        "width": "100%",
      }
    }
  />
</View>
`;

exports[`renders correctly StickersImage with earth image 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <Image
    resizeMode="contain"
    source={1}
    style={
      {
        "alignSelf": "center",
        "height": 650,
        "marginVertical": 20,
        "maxWidth": 320,
        "width": "100%",
      }
    }
  />
</View>
`;

exports[`renders correctly StickersImage with restaurant image 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <Image
    resizeMode="contain"
    source={1}
    style={
      {
        "alignSelf": "center",
        "height": 650,
        "marginVertical": 20,
        "maxWidth": 320,
        "width": "100%",
      }
    }
  />
</View>
`;
